cranelift/codegen/src/isa/x64/inst/args.rs

```diff
@@ -337,6 +337,7 @@ pub enum SseOpcode {
     Addss,
     Addsd,
     Andps,
+    Andpd,
```

Comment:
Since `Andps` already exists, is there any point in adding `Andpd`?  Similar question for `Orpd`.  They are functionally identical, so the only reason for doing this is if we believe we'll get some kind of domain-crossing penalty by applying (eg) `Andps` to the output of some insn that generates `F64x2` rather than `F32x4`.  Reading the opt guide sec 3.5.2.3 "Bypass between execution domains" leads me to believe that `Andpd/Andps/Orpd/Orps` would all "live" in the "integer domain".

Reply:
Or maybe all in the FP domain (who knows).  Point is, I can't see that the implied 32x4 vs 64x2 differentiation would make any difference.

Reply:
Good question, I like having the properly typed names, but since the F64 variants may be one byte longer (because of 66-prefix)... That's a question for our Intel experts :-)

@abrown @jlb6740 Is there a preference to use e.g. `andps` over `andpd`, if we know the inputs are actually doubles? (See above reference to Intel's manual that Julian quoted.)

Reply:
> leads me to believe that Andpd/Andps/Orpd/Orps would all "live" in the "integer domain".

@julian-seward1, reading that section, I would rather think that `Andpd/Andps/Orpd/Orps` all live in the FP domain, not in the SIMD (a.k.a. "integer") domain. The example there switches between `ADDPS` (FP) and `PAND` (SIMD) and back to `ADDPS` (FP). So I would expect that to keep everything in the same domain, to follow the manual's example, we would need `ADDPS; ANDPS; ADDPS`. Or am I missing something?

> Since Andps already exists, is there any point in adding Andpd?

The question still stands, though, and since I don't know of any penalty like the one mentioned (and I believe packed single floats and packed double floats to be in the same FP domain), then the assembler could avoid knowing about another instruction. But what is the benefit of that? We can always choose `ANDPS` in the lowering if we are concerned about code size? I would leave it in so that lowering users have the option to use it if it feels more natural.

I think where I'm coming from (and I'm still developing this thought, so bear with me) is that at the assembler level, which is what we are talking about here, the best naming convention in the assembler code is one that matches the ISA naming convention 1:1. Instead of writing `Inst::XmmAbcDef(SseOpcode::Andps, ...)` (or its helper function) I am suggesting the abstraction should look more like `...::Andps(...)` so that we more directly think in the ISA namespace than in Cranelift's custom namespace. From that viewpoint, it makes sense to me to have more instruction variants available (e.g. `ANDPD`) to reduce friction during lowering. (I understand that's still not strongest argument for keeping `ANDPD` but I'm really thinking more about the ISA naming point... it was a bit frustrating to have to think in terms of an ad hoc convention instead when I was addding SIMD FP arithmetic support).